onnxruntime/core/platform/path_lib.h

```diff
@@ -178,11 +178,10 @@ inline OrtFileType DTToFileType(DWORD dwFileAttributes) {
   return OrtFileType::TYPE_REG;
 }
 inline std::string FormatErrorCode(DWORD dw) {
-  char* lpMsgBuf;
-  FormatMessageA(FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS, NULL, dw,
-                 MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), (LPSTR)&lpMsgBuf, 0, NULL);
+  char lpMsgBuf[1024 * sizeof(TCHAR)];
```

Comment:
How to ensure it's enough?

Reply:
It doesn't seem like there is a way. Looking at Windows system error codes, seems like the vast majority will fit in this amount of bytes though. And if its not enough, the docs make it seem like the returned buffer will still be null terminated.

Reply:
from the doc (https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-formatmessage): 
If the FORMAT_MESSAGE_ALLOCATE_BUFFER flag is not set, this parameter specifies the size of the output buffer, in TCHARs. If FORMAT_MESSAGE_ALLOCATE_BUFFER is set, this parameter specifies the minimum number of TCHARs to allocate for an output buffer.

The output buffer cannot be larger than 64K bytes.

Reply:
Should I allocate a buffer of 64K? Seems like a lot of memory to use for messages that look like this: https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes--0-499-

Reply:
Actually, I just tested to see how the api would behave for error messages that don't fit into the buffer and it returns a corrupted string. So I'll go ahead and allocate a 64K buffer on the heap.